source/extensions/filters/udp/udp_proxy/udp_proxy_filter.h

```diff
@@ -294,7 +301,14 @@ class HttpUpstreamImpl : public HttpUpstream, protected Http::StreamCallbacks {
     }
 
     void decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool end_stream) override {
-      bool is_valid_response = Http::CodeUtility::is2xx(Http::Utility::getResponseStatus(*headers));
+      auto status = Http::Utility::getResponseStatus(*headers);
+      bool is_valid_post = parent_.tunnel_config_.usePost() && Http::CodeUtility::is2xx(status);
+      // In connect-udp case, Envoy will transform the H2 headers to H1 upgrade headers.
```

Comment:
I knew Envoy did this for requests, but does it do it for responses too? Can you point me at the code that does that?

Reply:
Yeah so in the request path, if the headers are an upgrade, Envoy will expect them to be in ``H1`` form, and will transform to ``H2``, and save the upgrade type in a member, see here:
https://github.com/envoyproxy/envoy/blob/7479b6cbdde02186b2fdbad3137965f628d3f42f/source/common/http/http2/codec_impl.cc#L278

On the response path, if the upgrade was set, Envoy will transform the ``H2`` headers to ``H1`` and set the upgrade as a header, see:
https://github.com/envoyproxy/envoy/blob/7479b6cbdde02186b2fdbad3137965f628d3f42f/source/common/http/http2/codec_impl.cc#L534

Reply:
Thanks!